### DISCUSSION TITLE
NM solver: 0 results found for twig

### DISCUSSION THREAD

**Discussion by kathmcp** [phoebe-project/phoebe2#521] [Questions]
Hi, since I have updated to 2.3.53, when I run the Nelder-Mead solver, it keeps coming up with the error: 

Sun, 05 Sep 2021 05:55 SOLVER       WARNING received error while setting values: 0 results found for twig: 'None', {'context': 'constraint', 'uniqueid': 'lbvYkJlglLqDoUHhjpVdApMtjumdVB', 'check_visible': False, 'check_default': True, 'check_advanced': False, 'check_single': False}. lnprobability=-inf

It does complete the solver, but looking at the results, the values for requiv and incl are the same as the initial inputs of 1.05 and 90, so it doesn't appear to have done anything to them. I was wondering if this is related to the above error?

I don't know if it is related to the new release, but I didn't get this error before. However, I have now added another passband (so now have B, V and R) and a mesh dataset as well, so am not sure if this could be the problem. 

I also get this same error when I try to plot the residuals

As the Emcee solver takes days to run, it would be good to know if there is something I should change before setting it off again.
Thanks!
![NMsolver](https://user-images.githubusercontent.com/81473857/132119052-88a38029-99a2-4adf-a2d9-23f354dab71d.PNG)
![NMsolver2](https://user-images.githubusercontent.com/81473857/132119054-939928f0-ca1b-41ab-8515-eeb0b9f32920.PNG)
![NMsolverResid](https://user-images.githubusercontent.com/81473857/132119094-92dc02b7-7995-4652-8889-d38fd03309d2.PNG)
![NMsolverResid2](https://user-images.githubusercontent.com/81473857/132119095-5c4401db-97f5-4c02-b0e1-e9069df0c60f.PNG)

**Comment by kecnry** [comment]
That warning is definitely concerning.  Can you show me the output of `print(b.get_value('fit_parameters', expand=True))`?

Was this a bundle that you saved in an old version and then imported after updating phoebe?  If so, there is a chance some constraint isn't migrating correctly for contact binaries - please send me the bundle file (either here or by email) and any information about how you created it (which version(s) of phoebe, etc) and I'll try to track down what went wrong and get you back up and running.

As for the final error during plotting, you adopted the solution which changes the face values of the parameters, but you still need to run a forward-model with `run_compute` before you'll have a synthetic model available for plotting.

**Reply by kathmcp** [reply]
This was a bundle that I created in v 2.3.51 with Jupyter Notebook. I updated phoebe using pip in Ubuntu to v2.3.53 and then changed the first line of code in the file to update phoebe to v2.3.53. Then I added another light curve dataset and a mesh dataset. Then when I ran the Nelder-Mead I got those errors. I am running the Emcee solver just now (which seems to be happily running after 1.5days so far), so can't send that actual file, but I have done the same thing with a previous saved version of the file and attach that now for you. I haven't been able to run it though because my computer is busy with the Emcee solver, but the lines of code should be the same. Sorry I can't stop the solver and send you that particular file, but my thesis is due on the 14th so I am running out of time now! I hope that helps, thanks again :)
[CB_LD_pblum_merit_NM-Emcee-v4-BVR-irrad-ext-mesh-Kyle.zip](https://github.com/phoebe-project/phoebe2/files/7117325/CB_LD_pblum_merit_NM-Emcee-v4-BVR-irrad-ext-mesh-Kyle.zip)

**Reply by kathmcp** [reply]
Hi again, sorry I just noticed that I did do a run_compute before plotting the residuals, it's just that it wasn't included in the snip above, so maybe this is also related to the Nelder-Mead errors.

**Reply by kecnry** [reply]
Ok, can you send me the datafiles as well (anything you load with pandas) or the saved bundle so I can try to reproduce?   And just to clarify - you changed the notebook to use 2.3.53 and then ran the notebook from the top again?  Or did you just continue running cells later in the notebook after updating PHOEBE?

As long as emcee is working for you now, I'm not too worried and it may have just been a fluke.  But my concern would be if samples in MCMC are being rejected for the same error.  If that were the case, then emcee may be running but not really progressing.

**Reply by kathmcp** [reply]
How do I find and send the data files? Also, the Emcee solver is still running so I am not sure how to send you the saved bundle. Did you get the one I attached yesterday? If not, I can email it.

Yes, I changed the notebook to 2.3.53 and then ran the notebook again from the top.

I am slightly concerned about Emcee solver, as it came up with some errors and warnings within the first 25minutes, but there has been absolutely nothing since, and that was on Sunday morning. Normally, warnings and errors pop up every now and again. I think it is still running, from looking at the CPU usage in task manager, but it is possibly not progressing like you say. The first errors and warnings that it produced were the usual ones, but the last few are runtimewarnings as attached.

Given I am running short on time, do you think it would be better to start a whole new file starting with v2.3.53, and trying to run that? Or should I leave Emcee running and see what happens? I am not sure how long it will take, it took 3 days with only 2 lightcurve datasets and no mesh, and now it has 3 lightcurves and a mesh.

**Reply by kecnry** [reply]
You sent me the notebook file, but that notebook loads several csv files from you local filesystem that I do not have but need in order to run the notebook.

I can't give you good advice on whether to restart or not.  If you do, my advice would be to set the number of iterations at a fairly low number and you can always [continue the mcmc run](http://phoebe-project.org/docs/2.3/tutorials/emcee_continue_from.ipynb) if you want after it completes.  That way you're not left waiting for it to complete and left with nothing if it gets stuck or fails.  If you think the problem is with 2.3.53 and it worked in 2.3.51, you may want to consider rolling back to that version and just manually adopt the mean values instead of the median (which I believe was the reason you updated, if I remember right).

Good luck and sorry for the troubles!

**Reply by kathmcp** [reply]
Ok thanks, I think I will try going back to 2.3.51 as I am not convinced I will get anything from this Emcee run.  Here are the csv files, sorry, I should have realised!
[BFilterAllFluxTimeFluxErr2.csv](https://github.com/phoebe-project/phoebe2/files/7123284/BFilterAllFluxTimeFluxErr2.csv)
[RFilterAllFluxTimeFluxErr.csv](https://github.com/phoebe-project/phoebe2/files/7123285/RFilterAllFluxTimeFluxErr.csv)
[VFilterAllFluxTimeFluxErr.csv](https://github.com/phoebe-project/phoebe2/files/7123286/VFilterAllFluxTimeFluxErr.csv)

**Reply by kathmcp** [reply]
Hi Kyle, I have stopped the Emcee solver now, so here is the bundle. Hope it helps!
[CB_LD_pblum_merit_NM-Emcee-v4-BVR-irrad-ext-mesh-v1.zip](https://github.com/phoebe-project/phoebe2/files/7123682/CB_LD_pblum_merit_NM-Emcee-v4-BVR-irrad-ext-mesh-v1.zip)

**Reply by kathmcp** [reply]
So, I have gone back to my old file in v2.3.51, reran it exactly as it was, and it appeared to still work ok. I then added my R filter data and a mesh dataset, and it came up with the same errors again. I have now removed the mesh dataset and it appears to be working so far. Maybe that is of some help to you?

**Reply by kecnry** [reply]
Ok, that's very helpful information.  It may take me a little while to track down what is actually happening, but at least it sounds like you should be able to move forward with your analysis without the mesh dataset?

**Reply by kecnry** [reply]
And do you mean the warnings in Nelder-Mead or errors when plotting?

**Reply by kathmcp** [reply]
I mean the warnings in Nelder-Mead when it said 0 results found for twig. This is not appearing anymore since I removed the mesh dataset.

Yes, I can continue on now, hopefully with my final run! I will add in a plot from the mesh dataset from a previous file, as it will look good in my report :)

**Reply by kecnry** [reply]
I've tried to recreate the bug you were seeing from scratch, but something is clearly different in your system that is necessary to reproduce the bug.  And it seems the notebooks you sent me don't raise the error as they are now.

So... when you get a chance (no rush - it can be after you finish your thesis), can you please save the bundle object (using `b_cb.save('export.bundle')` instead of the whole notebook), then try to create a separate short notebook/script that loads that bundle (`b_cb = phoebe.load('export.bundle')`) and reproduces the warnings/errors for Nelder-Mead?  It would be great if that short notebook could show it working without the mesh, then add the mesh and show that it starts failing.  If you can send me those two files (the bundle object and the short notebook showing just the bug itself in a few lines), then it'll be much easier to reproduce and track down the underlying issue.

Thanks and good luck wrapping up your thesis!

**Reply by kathmcp** [reply]
Hi Kyle, sorry for the delay in replying. I managed to hand my thesis in with half a day to spare :) I have taken a couple of weeks to recover and organise things, but I will start looking into this now. By the way, I gave you a special acknowledgement in my thesis, I literally couldn't have completed it without your help!

**Reply by kathmcp** [reply]
Hi again, I think this is what you need, but please let me know if there's anything else I can do.
[NMIssueMeshNoMesh.zip](https://github.com/phoebe-project/phoebe2/files/7266729/NMIssueMeshNoMesh.zip)

**Reply by kecnry** [reply]
Thanks, I'll take a look and try to track that down when I get a chance.  Congratulations on completing your thesis!
